méli-mélo/2023-09-menu/campaign-menu.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * @title Campaign menu
+ * @author PCH
+ */
+(function ($, window, document, wb) {
+    "use strict";
+
+    /*
+     * Variable and function definitions.
+     * These are global to the plugin - meaning that they will be initialized once per page,
+     * not once per instance of plugin on the page. So, this is a good place to define
+     * variables that are common to all instances of the plugin on a page.
+     */
+    var componentName = "campaign-menu",
+        selector = "." + componentName + ".gcweb-menu",
+        initEvent = "wb-init" + selector,
+        $document = wb.doc,
+        /**
+         * @method init
+         * @param {jQuery Event} event Event that triggered the function call
+         */
+        init = function (event) {
+            
+            var testing = true;
+
+            // Start initialization
+            // returns DOM object = proceed with init
+            // returns undefined = do not proceed with init (e.g., already initialized)
+            var elm = wb.init(event, componentName, selector),
+                $elm,
+                settings;
+
+            if (elm) {
+
+                // At this point, the GCWeb menu is already completed
+                $elm = $(elm);        
+                
+                // Check if there is already a gcweb menu (or if there are 2 present)
+                /// If global gcmenu is present, hide the campaign menu and return, do not continue
+                var gcWebMenus = document.querySelectorAll(".gcweb-menu");
+                if (gcWebMenus.length > 1) {
+                    console.warn(componentName + " - gcweb menu already exsits on the page, hiding gcweb campaign menu");
+                    $elm.addClass('hidden');
+                    return;
+                }
+                
+                // Apply appropriate visible CSS classes to the GCWeb menu
+                $elm.addClass( "visible-sm visible-xs" );
+                
+                // Use the GCWeb menu to create the megamenu for md and lg                
+                // If a megamenu is already present, abort to avoid duplicate wb-sm IDs 
+                var megamenuExists = document.querySelector("#wb-sm");
+                if (megamenuExists != undefined || megamenuExists != null) {
+                    console.warn(componentName + " can't start, megamenu already exsits on the page");
+                    wb.ready($elm, componentName);
+                    return;
+                }
```

Comment:
Testing note: 
Because we can't test this with gcweb-jekyll, I did manually tested into a wet-boew local dev environment and I confirm the plugin won't start if a mega-menu is present on the page. The warning is displayed as expected